invisible_cities/io/dst_io.py

```diff
@@ -24,3 +28,39 @@ def load_dst(filename, group, node):
 def load_dsts(dst_list, group, node):
     dsts = [load_dst(filename, group, node) for filename in dst_list]
     return pd.concat(dsts)
+
+
+def _store_pandas_as_tables(h5out:tb.file.File, df:pd.DataFrame, group_name:str, table_name:str, compression:str='ZLIB4', descriptive_string:Optional[str]=None, str_col_length:int=32)->None:
+    if len(df) == 0:
+        warnings.warn(f' dataframe is empty', UserWarning)
+    if '/'+group_name not in h5out:
+        group = h5out.create_group(h5out.root,group_name)
+    else:
+        group = getattr(h5out.root,group_name)
+    if table_name in group:
+        table=getattr(group,table_name)
+    else:
+        tabledef={}
+        for indx, colname in enumerate(df.columns):
+            coltype = df[colname].dtype.name
+            try:
+                tabledef[colname] = tb.Col.from_type(coltype, pos = indx)
+            except ValueError:
+                tabledef[colname] = tb.StringCol(str_col_length, pos = indx)
```

Comment:
I am not sure if `try/except` is optimal here. In principle, you could have a ValueError raised for a different reason. Also, I think I would like this to be a separate function, but I don't have a strong opinion.